src/core/src/sys/compute.rs

```diff
@@ -236,8 +236,8 @@ impl Compute for PressureAtTemperature {
         let virial_tensor = system.virial();
         let virial = virial_tensor.trace();
         let volume = system.volume();
-        let natoms = system.size() as f64;
-        return natoms * K_BOLTZMANN * self.temperature / volume + virial / (3.0 * volume);
+        let nmols = system.molecules().len() as f64;
```

Comment:
You mean why use the number of molecules instead of number of atoms? I can try to explain the idea in some sentences - if you want, I can also send you a derivation via mail.

The idea is, that for a system consisting of noninteracting molecules (no intermolecular interactions but generic intramolecular potentials) we can write the NVT partition function as a product of single-molecule partition functions. 

We can further do a coordinate transformation such that the positions of the atoms of a molecule (let's say that there are `n` atoms per molecule) are reformulated to yield the center of mass position of the molecule plus the `n-1` *relative positions* in the molecule. The intra molecular energy of a single molecule does not depend on the center of mass position of the molecule, so we can perform the integration over the center of mass position in the partition function directly. This integral yields `V` (the volume) for every molecule, i.e. for `N` molecules this is `V^N`.

The rest of the partition function is no longer a function of the volume, but of temperature and the number of atoms in the molecule. The pressure is computed by taking the derivative of the Helmholtz energy w.r.t the volume. The only volume dependent term of the ideal gas partition function of a molecular system is `V^N`. Since we have to take the logarithm of the partition function, we get `p = kT * d/dV (N log V) = NkT/V`.

Does that make sense? It's a bit difficult to describe in words.